engine/core/src/main/java/org/datacleaner/job/builder/AbstractComponentBuilder.java

```diff
@@ -383,10 +424,62 @@ protected boolean setConfiguredPropertyIfChanged(final ConfiguredPropertyDescrip
             }
         }
 
+        synchronizeDependentProperties(configuredProperty, value, currentValue);
+
         configuredProperty.setValue(_configurableBean, value);
         return true;
     }
 
+    private void synchronizeDependentProperties(final ConfiguredPropertyDescriptor property,
+            final Object newValue, final Object currentValue) {
+        if (currentValue != null) {
+            getDescriptor().getConfiguredPropertiesByAnnotation(MappedProperty.class).stream().filter(
+                    dependentProperty -> property.getName().equals(dependentProperty.getAnnotation(MappedProperty.class)
+                            .value())).forEach(dependentProperty -> {
+                                // In case the new value no longer contains everything in the original value,
+                                // the values in the dependent property referring to the removed values need
+                                // to be removed too.
+                                Object dependentValue = dependentProperty.getValue(_configurableBean);
+
+                                if (dependentValue != null) {
+                                    // First build a list containing value and references tuples.
+                                    Map<Integer, Object> originalMappings = new HashMap<>();
+
+                                    List<Object> synchronizedDependents = new ArrayList<>();
+
+                                    if (currentValue.getClass().isArray()) {
+                                        for (int i = 0; i < Array.getLength(currentValue); i++) {
+                                            originalMappings.put(Array.get(currentValue, i).hashCode(), Array.get(
+                                                    dependentValue, i));
+                                        }
+
+                                        for (int i = 0; i < Array.getLength(newValue); i++) {
+                                            synchronizedDependents.add(originalMappings.get(Array.get(newValue, i)
+                                                    .hashCode()));
+                                        }
+
+                                        dependentProperty.setValue(_configurableBean, getArray(dependentProperty
+                                                .getBaseType(), synchronizedDependents));
+                                    } else {
+                                        if (newValue == null) {
+                                            dependentProperty.setValue(_configurableBean, null);
+                                        }
+                                    }
+                                }
+                            });
+        }
+    }
+
+    private static <E> E[] getArray(Class<E> clazz, List<?> baseList) {
+        E[] result = (E[]) Array.newInstance(clazz, baseList.size());
```

Comment:
I will add the final :), this method is really needed, because I have a list of objects and I need to pass a typed array to the ConfiguredPropertyDescriptor#setValue(Object, Object) method, or otherwise I'll get some exception that the value can't be set because it is of the wrong type. I could figure out no easier way to get a typed array from my list of objects (believe me, I tried).